src/box/iproto.cc

```diff
@@ -3299,6 +3299,42 @@ iproto_set_msg_max(int new_iproto_msg_max)
 	}
 }
 
+int
+iproto_session_send(struct session *session,
+		    const char *header, const char *header_end,
+		    const char *body, const char *body_end)
+{
+	assert(session->type == SESSION_TYPE_BINARY);
+	struct iproto_connection *con =
+		(struct iproto_connection *)session->meta.connection;
+	if (con->state != IPROTO_CONNECTION_ALIVE) {
+		diag_set(ClientError, ER_SESSION_CLOSED);
+		return -1;
+	}
+
+	ptrdiff_t header_size = header_end - header;
+	ptrdiff_t body_size = body_end - body;
+	ptrdiff_t packet_size = 5 + header_size + body_size;
+	char *buf = (char *)obuf_alloc(con->tx.p_obuf, packet_size);
+	if (buf == NULL) {
+		diag_set(OutOfMemory, packet_size, "obuf_alloc", "buf");
+		return -1;
+	}
+	char *p = buf;
+	*(p++) = INT8_C(0xce);
+	p = mp_store_u32(p, packet_size - 5);
```

Comment:
This snippet was discussed with 5 people already :joy:. All discussions came to a conclusion that we should stick with the format of IPROTO packets sent by Tarantool servers (i.e., the fixed 5-byte size of the packets) — inter alia, it simplifies debugging of dumped IPROTO packets.

Reply:
In the older comments I haven't seen any explanation why it was 5 bytes. So here I gave it. I don't mind keeping it 5 bytes. Although it could be better if someone would extract it into a function with a comment explaining why it is 5 bytes. So as people in the future wouldn't ask that again.